src/clj/rems/application/model.clj

```diff
@@ -18,82 +18,74 @@
+(def ^:private submittable-application-commands
+  [:application.command/save-draft
+   :application.command/submit
+   :application.command/close
+   :application.command/remove-member
+   :application.command/invite-member
+   :application.command/uninvite-member
+   :application.command/accept-licenses
+   :application.command/change-resources])
+
+(def ^:private non-submittable-application-commands
+  [:application.command/remove-member
+   :application.command/uninvite-member
+   :application.command/accept-licenses])
+
+(def ^:private handler-all-commands
+  #{:application.command/remark
+    :application.command/add-licenses
+    :application.command/add-member
+    :application.command/change-resources
+    :application.command/remove-member
+    :application.command/invite-member
+    :application.command/uninvite-member
+    :application.command/request-comment
+    :application.command/request-decision
+    :application.command/return
+    :application.command/approve
+    :application.command/reject})
+
+(def ^:private handler-returned-commands
+  (disj handler-all-commands :application.command/return))
+
+(def ^:private created-permissions
+  {:applicant submittable-application-commands
    :member [:application.command/accept-licenses]
-   :handler [:see-everything
-             :application.command/remark
-             :application.command/remove-member
-             :application.command/uninvite-member]
-   :commenter [:see-everything
-               :application.command/remark]
-   :decider [:see-everything
-             :application.command/remark]
-   ;; roles whose permissions don't change
    :reporter [:see-everything
               :application.command/remark]
-   :past-commenter [:see-everything
-                    :application.command/remark]
-   :past-decider [:see-everything
-                  :application.command/remark]
    ;; member before accepting an invitation
    :everyone-else [:application.command/accept-invitation]})
 
 (def ^:private submitted-permissions
-  {:applicant [:application.command/remove-member
-               :application.command/uninvite-member
-               :application.command/accept-licenses]
-   :member [:application.command/accept-licenses]
-   :handler [:see-everything
-             :application.command/remark
-             :application.command/add-licenses
-             :application.command/add-member
-             :application.command/change-resources
-             :application.command/remove-member
-             :application.command/invite-member
-             :application.command/uninvite-member
-             :application.command/request-comment
-             :application.command/request-decision
-             :application.command/return
-             :application.command/approve
-             :application.command/reject]
+  {:applicant non-submittable-application-commands
+   :handler (vec (conj handler-all-commands :see-everything))
```

Comment:
These vec calls are not needed. update-role-permissions doesn't care what kind of a collection the parameter is.

Reply:
Ok, should I then change all these vectors to sets for consistency?

Reply:
Yes, sounds good.

Reply:
Done.